core/blockchain_mmr.go

```diff
@@ -0,0 +1,242 @@
+package core
```

Comment:
Can I ask you to please add a few more comments above function definitions in this file?